### PR TITLE
job to delete trashed ciphers nightly

### PR DESCRIPTION
**Pull request by kspearrin**
This PR adds a job to delete ciphers that have been previously "trashed" over 30 days ago. The job will run at midnight, nightly, running a sproc that will delete any items that are older than 30 days ago.

### CODE REVIEW

File: src/Core/Repositories/SqlServer/CipherRepository.cs [hunk 1]
```diff
@@ -624,6 +624,18 @@ public async Task<DateTime> RestoreAsync(IEnumerable<Guid> ids, Guid userId)
             }
         }
 
+        public async Task DeleteDeletedAsync(DateTime deletedDateBefore)
```
Comment (addisonbeck) [inline src/Core/Repositories/SqlServer/CipherRepository.cs:627]: "DeleteDeleted": I love it

File: src/Admin/Jobs/JobsHostedService.cs [hunk 2]
```diff
@@ -55,13 +55,19 @@ public override async Task StartAsync(CancellationToken cancellationToken)
...
+                .Build();
 
             var jobs = new List<Tuple<Type, ITrigger>>
             {
                 new Tuple<Type, ITrigger>(typeof(DeleteSendsJob), everyFiveMinutesTrigger),
                 new Tuple<Type, ITrigger>(typeof(DatabaseExpiredGrantsJob), everyFridayAt10pmTrigger),
                 new Tuple<Type, ITrigger>(typeof(DatabaseUpdateStatisticsJob), everySaturdayAtMidnightTrigger),
-                new Tuple<Type, ITrigger>(typeof(DatabaseRebuildlIndexesJob), everySundayAtMidnightTrigger)
+                new Tuple<Type, ITrigger>(typeof(DatabaseRebuildlIndexesJob), everySundayAtMidnightTrigger),
+                new Tuple<Type, ITrigger>(typeof(DeleteCiphersJob), everyDayAtMidnightUtc)
```
Comment (addisonbeck) [inline src/Admin/Jobs/JobsHostedService.cs:70]: Are the other jobs not based on UTC time? This is the only one specifying such in the trigger name.

Comment (kspearrin) [reply src/Admin/Jobs/JobsHostedService.cs:70]: The others are in Eastern time zone. This is the only UTC one, which is why I labeled it this way

**Review by addisonbeck — DISMISSED** [review]

**Comment by M09NST3R** [comment]
@addisonbeck @kspearrin 

I think you should consider giving people more time to rescue items from their Trash folders first, and ideally do some user education as well.

Without looking at the code, I understand that this PR would immediately delete everything that has been in Trash for more than 30 days once the script is first executed, correct? This might lead to very unfortunate and unexpected outcomes for some folks.

AFAIK, there's currently no warning in the UI that the trash gets cleared automatically after a certain number of days (at least I couldn't see it in the desktop app), and there's also no straight forward way for people to make a backup of their Trash folders.

**Comment by cscharf** [comment]
@M09NST3R , we're going to consider setting a future "target date" for this job to start from to give everyone enough warning/heads' up along with some messages, banners and notices in the broader community.

**Comment by kspearrin** [comment]
Added an admin setting so that we can turn this off and back on again whenever we feel comfortable after release. Upon release, we need to set this new setting to some high value so that the deletes don't actually occur yet.

**Review by cscharf — APPROVED** [review]
Send some comments via slack as well Re: deployment activities, but looks good with the updates.

**Pull request merged**